src/core/thread/thread_netif.cpp

```diff
@@ -122,17 +125,23 @@ Error ThreadNetif::SendMessage(Message &aMessage) { return Get<MeshForwarder>().
 
 Error ThreadNetif::RouteLookup(const Ip6::Address &aSource, const Ip6::Address &aDestination)
 {
-    Error    error;
+    Error    error = kErrorNone;
     uint16_t rloc;
 
-    SuccessOrExit(error = Get<NetworkData::Leader>().RouteLookup(aSource, aDestination, rloc));
+    error = Get<NetworkData::Leader>().RouteLookup(aSource, aDestination, rloc);
 
-    if (rloc == Get<Mle::MleRouter>().GetRloc16())
+    if (error == kErrorNone)
+    {
+        if (rloc == Get<Mle::MleRouter>().GetRloc16())
+        {
+            error = kErrorNoRoute;
+        }
+    }
+    else
     {
-        error = kErrorNoRoute;
+        LogWarn("Failed to find the valid route for: %s", aDestination.ToString().AsCString());
```

Comment:
I agree it is useful to log this, but I feel `warn` may be too high? Maybe we can use `note` or `info` level?

The equivalent existing logs to this that I can think of, are the `MeshForwarder::LogMessage()` logs (when a message tx fails e.g. not acked, etc and we log message drop or errors). We use `note` for all failures in `MeshFowarder` and `Info` to log success tx.